river/linear_model/glm.py

```diff
@@ -94,6 +157,13 @@ def _eval_gradient_one(self, x: dict, y: float, w: float) -> (dict, float):
                 loss_gradient * utils.VectorDict(x) + self.l2 * self._weights,
                 loss_gradient,
             )
+
+        if self.l1:
+            return (
+                loss_gradient * utils.VectorDict(x),
+                loss_gradient,
+            )
+
```

Comment:
I don't think we need to add `if self.l1` here, since it would return `(loss_gradient * utils.VectorDict(x), loss_gradient)` anyways.